File: concrete/src/test/java/pcb/uwu/amount/composite/electromagnetism/TeslasTest.java
package pcb.uwu.amount.composite.electromagnetism;

import org.junit.Test;
import pcb.uwu.amount.composite.fundamental.Area;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.composite.fundamental.SquareMeterUnit.SQUARE_METER;

public class TeslasTest {
	private static final Teslas teslas = new Teslas(2);

	@Test
	public void multipliedByArea() {
		assertEquals(new Webbers(4),
				teslas.multipliedBy(new Area(2, SQUARE_METER), DECIMAL64));
	}

}

File: implementation/src/test/java/pcb/uwu/amount/derived/electromagnetism/TeslasTest.kt
package pcb.uwu.amount.derived.electromagnetism

import pcb.uwu.amount.derived.geometry.Area
import pcb.uwu.unit.derived.geometry.SquareMeterUnit
import kotlin.test.Test
import kotlin.test.assertEquals

class TeslasTest
{
    @Test
    fun multipliedByArea() =
        assertEquals(expected = Webbers(4),
                     actual = Teslas(2) * Area(amount = 2, unit = SquareMeterUnit))

}